autotest/gdrivers/fits.py

```diff
@@ -82,3 +81,34 @@ def fits_metadata():
     ds2 = None
 
     assert md['TEST2'] == 'test_value2'
+
+def test_fits_nodata():
+    driver = gdal.GetDriverByName('FITS')
+
+    ds = gdal.Open('../gcore/data/nodata_byte.tif')
+    ds2 = driver.CreateCopy('tmp/nodata_byte.fits', ds)
+    ds2 = None
+    gdal.Unlink('tmp/nodata_byte.fits.aux.xml')
+
+    ds2 = gdal.Open('tmp/nodata_byte.fits')
+    nd = ds2.GetRasterBand(1).GetNoDataValue()
+    ds2 = None
+
+    assert nd == 0
+
+def test_fits_offscale():
+    driver = gdal.GetDriverByName('FITS')
+
+    ds = gdal.Open('../gcore/data/offscale_byte.tif')
+    ds2 = driver.CreateCopy('tmp/offscale_byte.fits', ds)
+    ds2 = None
+    gdal.Unlink('tmp/offscale_byte.fits.aux.xml')
+
+    ds2 = gdal.Open('tmp/offscale_byte.fits')
```

Comment:
the file should be Unlink() after use

Reply:
Ok, but then I realize that I did'nt understand the difference between `gdal.Unlink('myfile.fits')`
and `driver.Delete('myfile.fits')` used e.g. in the [first test](https://github.com/OSGeo/gdal/blob/35bb75627ee70aa42fd1e1d80d1af485103fa475/autotest/gdrivers/fits.py#L55).
Which one is more appropriate in that case?

Reply:
driver.Delete() delete the file and its auxiliary files like .aux.xml. Using it is safe

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/8e48b80f479616a461e1cbe590cdd4e32baed012